### ISSUE TITLE
Duplicate Type warning when in Debug mode

### ISSUE THREAD

**Issue by Poolshark** [wp-graphql/wp-graphql#1769]
If I activate the debug log and execute any query, I get the warning that `DUPLICATE_TYPES` warning for two types which are defined in WPGraphQL:

- CoverTemplate
- FullWidthTemplate

Here's the full stack:

**Query**
```
query MyQuery {
  users {
    edges {
      node {
        id
      }
    }
  }
}
```

**Response**
```
{
  "data": {
    "users": {
      "edges": [
        {
          "node": {
            "id": "dXNlcjoy"
          }
        },
        {
          "node": {
            "id": "dXNlcjox"
          }
        }
      ]
    }
  },
  "extensions": {
    "debug": [
      {
        "type": "DUPLICATE_TYPE",
        "message": "You cannot register duplicate Types to the Schema. The Type 'CoverTemplate' already exists in the Schema. Make sure to give new Types a unique name.",
        "type_name": "CoverTemplate",
        "stack": [
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/TypeRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/access-functions.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/plugin.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/TypeRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/plugin.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/TypeRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/SchemaRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/WPGraphQL.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Request.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Router.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Router.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/plugin.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/functions.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-blog-header.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/index.php"
        ]
      },
      {
        "type": "DUPLICATE_TYPE",
        "message": "You cannot register duplicate Types to the Schema. The Type 'FullWidthTemplate' already exists in the Schema. Make sure to give new Types a unique name.",
        "type_name": "FullWidthTemplate",
        "stack": [
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/TypeRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/access-functions.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/plugin.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/TypeRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/plugin.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/TypeRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Registry/SchemaRegistry.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/WPGraphQL.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Request.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Router.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-content/plugins/wp-graphql/src/Router.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp-hook.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/plugin.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/class-wp.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-includes/functions.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/wp-blog-header.php",
          "/Users/florianragossnig/Local Sites/wp-graphql-plugin/app/public/index.php"
        ]
      }
    ]
  }
}
```

Wordpress Version: 5.7
WP GraphQL Version: 1.2.5

Thanks!

**Comment by jasonbahl** [comment]
@Poolshark Are these templates you have defined on your site like so: https://developer.wordpress.org/themes/template-files-section/page-template-files/#creating-page-templates-for-specific-post-types?

If not, are these Custom Post Types or Custom Fields or something else?

**Comment by Poolshark** [comment]
Hi! Thank you for the quick answer.

I was doing quite a few tests to reproduce this problem and it seems that it had something to do with an old version of the TentyTwenty Theme which I had activated for testing purposes. I have set up various clean WP installs with WP GraphQL and tried different themes. I could not reproduce the warning again.

Sorry for the false alarm!

**Comment by jasonbahl** [comment]
@Poolshark I'll do a little digging. I think it might be a legit bug.

**Comment by Poolshark** [comment]
So I have done some more testing but I could not reproduce the problem anymore. The problem disappeared with updating the theme **TwentyTwenty** to the currently newest version `v 1.7`.

Any news from your side?

**Comment by jasonbahl** [comment]
@Poolshark my theory was that if I added multiple templates and assigned them to different post types that there might have been a bug that was registering the template Type to the Schema multiple times. 

I tried to replicate this by adding the following template files: 

```php
// test-template.php
<?php
/**
 * Template Name: Test Template
 * Template Post Type: post, page
 */
 ```
 
 And 
 
 ```php
 // test-template-2.php
 <?php
/**
 * Template Name: Test Template 2
 * Template Post Type: post, page
 */
```

I also filtered the Post post_type to support page-attributes, like so:

```php
add_filter( 'register_post_type_args', function( $args, $post_type ) {

	if ( 'post' === $post_type ) {
		$args['supports'][] = 'page-attributes';
	}
	
	return $args;

}, 10, 2 );
```

And my theory was that since the templates are assigned to multiple post types there might have been an error in logic that maps the templates to the Schema that was causing the duplicate type bug. 

But, even with these changes no duplicate type warnings, so my theory is not accurate. 

I'll close this issue as we're not able to replicate.
